machine_protocol.c

```diff
@@ -90,6 +95,14 @@ void protocol_init(PROTOCOL_STAT *s){
     s->allow_ascii = 1;
     s->send_serial_data = nosend;
     s->send_serial_data_wait = nosend;
+
+    // Check if all lengths in params can actually be received
+    for (int i = 0; i < paramcount; i++) {
+        if( params[i].len > sizeof( ((PROTOCOL_BYTES_WRITEVALS *)0)->content ) ) {
+            params[i].len = 0;  // Set to 0, so nothing can be received
+            while(1) {};        // Crash the processor intentionally.
```

Comment:
Crashing might be too harsh..

Reply:
setting len to zero is enough to indicate error?
Or setting length to sizeof( ((PROTOCOL_BYTES_WRITEVALS *)0)->content ) may be enough 'protection'.
Or maybe we should have a string which is (if filled) sent regularly on console.  Then in this circumstance, we could set the string to 'You Broke Me', and we'd all know.... (as above code is too early to spit one debug message out?).

Reply:
My preferred solution is to set it to zero, otherwise you might have some things working but not others. Debugging will be painful..

I was thinking about adding a return value to protocol_init. That way you can decide based on your application what to do, when init failed.